File: DemoMethods/src/main/java/net/ravendb/demo/basic/Query3.java
package net.ravendb.demo.basic;

import net.ravendb.abstractions.basic.Reference;
import net.ravendb.client.IDocumentSession;
import net.ravendb.client.RavenQueryStatistics;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Company;
import net.ravendb.demo.entities.QCompany;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@Controller
public class Query3 {
    @RequestMapping("/Basic/Query3")
    public Company query3(@RequestParam(value = "city", defaultValue = "Berlin") String city) {
        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {
            Reference<RavenQueryStatistics> statsRef = new Reference<>();

            QCompany c = QCompany.company;

            List<Company> query = session.query(Company.class)
                    .statistics(statsRef)
                    .where(c.address().city.eq(city))
                    .toList();

            Company company = query.stream().findFirst().orElse(null);
            return company;
        }
    }
}

File: DemoMethods/PythonDemo/demo_methods/Basic/Query3.py
from demo.holder import DocumentStoreHolder
from demo.entities import *


class Query3(object):
    @staticmethod
    def query3(city="Berlin"):
        with DocumentStoreHolder.get_store().open_session() as session:
            query, stats = list(session.query(object_type=Company, with_statistics=True,
                                              nested_object_types={"Address": AddressC,
                                                                   "Contact": ContactC}).
                                where_equals("Address.City", city))

            return query[0] if query else None
